Update javadsl sample (#129)
spring-cloud/spring-cloud-dataflow-samples: src/main/asciidoc/javadsl/main.adoc

		return propertiesBuilder.build();
	}
----

> ==== Step 2 Run a local Data Flow Server and run the sample application.

Reviewer: ```suggestion
==== Step 2 Start Data Flow and run the sample application
```

2017-11-22 18:04:08.647  INFO 26655 --- [r.woodchuck-1-1] log-sink          : wood
----
Note that the partitioning is done based on the hash of the `java.lang.String` object.

==== Step 5 Use Authentication

> Optionally if you have enabled authentication on a server there are 3

Reviewer: ```suggestion
Optionally, if you have enabled authentication in SCDF, there are three
```

> different ways to authorize a client.

Reviewer: ```suggestion
different ways to authorize the sample application (i.e. the client).
```

> Use basic authentication:

Reviewer: Could we maybe include the `current` link of SCDF's security docs that list these 3 options?

Author: Looks like we don't have any docs as in shell we created and documented those shortcuts like `--dataflow.username`. Oauth option were added just for tile and never documented. Java DSL just points to microsite and we don't have anything there either.

[source,bash,options="nowrap"]
----
$ java -jar target/scdfdsl-0.0.1-SNAPSHOT.jar \
  --spring.cloud.dataflow.client.authentication.basic.username=user \
  --spring.cloud.dataflow.client.authentication.basic.password=password
----

> Use OAuth client settings:

Reviewer: ```suggestion
Use OAuth client settings (UAA is used as the identity provider in this sample):
```

[source,bash,options="nowrap"]
----
$ java -jar target/scdfdsl-0.0.1-SNAPSHOT.jar \
  --spring.cloud.dataflow.client.authentication.client-id=dataflow \
  --spring.cloud.dataflow.client.authentication.client-secret=secret \
  --spring.cloud.dataflow.client.authentication.token-uri=http://localhost:8080/uaa/oauth/token \
  --spring.cloud.dataflow.client.authentication.scope=dataflow.create,dataflow.deploy,dataflow.destroy,dataflow.manage,dataflow.modify,dataflow.schedule,dataflow.view
----

Use OAuth access token:
[source,bash,options="nowrap"]
----
$ java -jar target/scdfdsl-0.0.1-SNAPSHOT.jar \
  --spring.cloud.dataflow.client.authentication.access-token=849228ed663e450ab5051c998eb71a4a
----

> For example with _uaa_ access token can be requested with:

Reviewer: ```suggestion
For example, if you're using UAA as the identity provider backend, the access token can be requested with the following command:
```
